src/online2bin/online2-wav-nnet3-wake-word-decoder-faster.cc

```diff
@@ -0,0 +1,288 @@
+
+        nnet3::DecodableAmNnetLoopedOnline decodable(trans_model,
+            decodable_info, feature_pipeline.InputFeature(),
+            feature_pipeline.IvectorFeature());
+
+        OnlineWakeWordFasterDecoder decoder(*decode_fst, decoder_opts,
+                                            silence_phones, trans_model);
+        OnlineTimer decoding_timer(utt);
+
+        BaseFloat samp_freq = wave_data.SampFreq();
+        int32 chunk_length;
+        if (chunk_length_secs > 0) {
+          chunk_length = int32(samp_freq * chunk_length_secs);
+          if (chunk_length == 0) chunk_length = 1;
+        } else {
+          chunk_length = std::numeric_limits<int32>::max();
+        }
+
+        int32 samp_offset = 0;
+
+        bool partial_res = false;
+        decoder.InitDecoding();
+        while (samp_offset < data.Dim()) {
+          int32 samp_remaining = data.Dim() - samp_offset;
+          int32 num_samp = chunk_length < samp_remaining ? chunk_length
+                                                         : samp_remaining;
+
+          SubVector<BaseFloat> wave_part(data, samp_offset, num_samp);
+          feature_pipeline.AcceptWaveform(samp_freq, wave_part);
+
+          samp_offset += num_samp;
+          decoding_timer.WaitUntil(samp_offset / samp_freq);
+          if (samp_offset == data.Dim()) {
+            // no more input. flush out last frames
+            feature_pipeline.InputFinished();
+          }
+
+          decoder.AdvanceDecoding(&decodable);
+          if (decodable.IsLastFrame(decoder.NumFramesDecoded() - 1)) {
+            std::vector<int32> word_ids;
+            decoder.FinishTraceBack(&out_fst);
+            fst::GetLinearSymbolSequence(out_fst,
+                                         static_cast<vector<int32> *>(0),
+                                         &word_ids,
+                                         static_cast<LatticeArc::Weight*>(0));
+            PrintPartialResult(word_ids, word_syms, partial_res || word_ids.size());
+            partial_res = false;
+            decoder.GetBestPath(&out_fst);
+            std::vector<int32> tids;
+            fst::GetLinearSymbolSequence(out_fst,
+                                         &tids,
+                                         &word_ids,
+                                         static_cast<LatticeArc::Weight*>(0));
+            //if (!word_ids.empty())
+            words_writer.Write(utt, word_ids);
+            alignment_writer.Write(utt, tids);
+            break;
+          } else {
+            std::vector<int32> word_ids;
+            if (decoder.PartialTraceback(&out_fst)) {
+              fst::GetLinearSymbolSequence(out_fst,
+                                           static_cast<vector<int32> *>(0),
+                                           &word_ids,
+                                           static_cast<LatticeArc::Weight*>(0));
+              PrintPartialResult(word_ids, word_syms, false);
```

Comment:
what does the output of this program look like?  Surely there normally wouldn't be  anything to print out.

Reply:
It prints to the decoding log like:
`
LOG (online2-wav-nnet3-wake-word-decoder-faster[5.5.500~1-247c0]:main():online2-wav-nnet3-wake-word-decoder-faster.cc:273) Decoded utterance
078a4bd56cc351e5a9a4eb4dc7ffc007-df0bf3f7-7552-40c7-aebd-6d99f1476300
`
`
<sil> HeySnips <sil>
`